src/plugins/vis_type_vega/public/data_model/ppl_parser.ts

```diff
@@ -34,6 +55,11 @@ export class PPLQueryParser {
       );
     }
 
+    if (timefield) {
+      const query = this.injectTimeFilter(url.body.query, timefield);
+      url.body.query = query;
```

Comment:
Is this the url that vega uses to query the cluster? I wouldn't expect the url parser to format/mutate the url, but I'm new to this area so it may be a gap in my understanding.

Reply:
> Is this the url that vega uses to query the cluster? 

Yes, `url.body.query` is the ppl query and will be send to opensearch. We modify it here to encapsulate the time filter from the UI, for example, the time range user selected on a dashboard. The behavior is the same as vega visualization with opensearch DSL which modified the query DSL to add time filter.

> I wouldn't expect the url parser to format/mutate the url

Could you please share your concern?

Reply:
That makes sense. No concerns, my expectation was that the parser is a pure function but it looks like this is more of a parser and formatter of the original url if I'm understanding correctly.

Reply:
I got your point now, I think we can also make it a pure function, for example we can deep copy the `url` object and then modify the deep copied object. I don’t notice much of a difference in this particular case since the existing parser mutate the url object as well, but I’d love to hear your thoughts on it :)